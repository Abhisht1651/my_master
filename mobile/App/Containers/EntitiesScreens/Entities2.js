// @flow
import { NavigationContainerProps } from '@react-navigation/native'
import React, { useState } from 'react'
import { View, Text, TouchableOpacity, Image, ScrollView, ImageProps } from 'react-native'
import images from '../../Themes/Images'
import Bubbles2 from '../../Components/Icons/Bubbles2'
import ThumbsLarge from '../../Components/Entities/ThumbsLarge'
import Profile from '../../Components/Entities/Profile'
import Storyline from '../../Components/StoryLineShortCardDefualt'
import FilterSVG from '../../Images/Icons/Icon-filter.svg'
import TimeFilter from '../../Components/TimeFilter/TimeFilter'
import StoryLineLongFormMultipleEvents from '../../Components/StoryLineLongFormMultipleEvents'

// Styles
import styles from './ScreenStyles/Entities2Styles'
import { X2SButton } from '../../Components/Buttons/X2SButton'

type Props = {
  title: string,
  about: string,
  count: number,
  details: string,
  profilePic: ImageProps
} & NavigationContainerProps

const Entities2 = (props: Props) => {
  const { navigation } = props
  const [TimeFilterPopUpVisible, setTimeFilterPopUpVisible] = useState(false)
  const openTimeFilterPopUp = () => {
    setTimeFilterPopUpVisible(true)
  }
  const closeTimeFilterPopUp = () => {
    setTimeFilterPopUpVisible(false)
  }
  const captureYears = (arr: Array<string>) => {
    console.log(arr)
  }
  const captureMonths = (arr: Array<string>) => {
    console.log(arr)
  }
  const pageNavigation = (navigate) => {
    navigation.navigate(navigate)
  }
  const Entities2Navigation = () => {
    navigation.navigate('Entities2')
  }
  const [pressStatus, setStatus] = useState(false)
  const changeStatus = () => {
    console.log('Hi')
    setStatus(!pressStatus)
  }
  return (
    <View style={styles.screen}>
      <View style={styles.topBox}>
        <TouchableOpacity onPress={() => navigation.goBack()}>
          <Image source={images.arrow} />
        </TouchableOpacity>
        <Bubbles2 />
      </View>
      <ScrollView>
        <View style={styles.contentBox}>
          <View style={styles.smallContentBox}>
            <View style={styles.profilepicBox}>
              <Profile profileIcons={[props.profilePic]} />
            </View>
            <View style={styles.profileDetails}>
              <Text style={styles.profileTitleFont}>{props.title}</Text>
              <Text style={styles.profileAboutFont}>{props.about}</Text>
            </View>
            <View style={styles.followContentBox}>
              <TouchableOpacity onPress={changeStatus}>
                {!pressStatus ? (
                  <X2SButton outline title="Follow" onPress={changeStatus} />
                ) : (
                  <X2SButton disabled title="Following" onPress={changeStatus} />
                )}
              </TouchableOpacity>
              <Text style={styles.followDetails}>{`${props.count}K Following`}</Text>
            </View>
            <View style={styles.profileDetailsBox}>
              <Text style={styles.detailTitleFont}>Details</Text>
              <Text style={styles.detailsFont}>{props.details}</Text>
            </View>
          </View>
        </View>

        <View style={styles.footerBox}>
          <View style={styles.footerBox1}>
            <View style={styles.textBox}>
              <Text style={styles.textFont1}>Similar Entities</Text>
              <TouchableOpacity onPress={() => navigation.navigate('RelatedEntities')}>
                <Text style={styles.textFont2}>View All</Text>
              </TouchableOpacity>
            </View>
          </View>
          <View style={styles.footerBox2}>
            <ScrollView horizontal>
              <View style={styles.box2}>
                <ThumbsLarge title="India" profileIcons={[images.EntitiesImage0]} />
              </View>
              <View style={styles.box}>
                <ThumbsLarge title="USA" profileIcons={[images.EntitiesImage3]} />
              </View>
              <View style={styles.box} profileIcons={[images.EntitiesImage4]}>
                <ThumbsLarge title="UNICEF" profileIcons={[images.EntitiesImage4]} />
              </View>
              <View style={styles.box}>
                <ThumbsLarge title="European" profileIcons={[images.EntitiesImage6]} />
              </View>
              <View style={styles.box}>
                <ThumbsLarge title="Europe" profileIcons={[images.EntitiesImage5]} />
              </View>
              <View style={styles.box}>
                <ThumbsLarge />
              </View>
              <View style={styles.box}>
                <ThumbsLarge />
              </View>
            </ScrollView>
          </View>
        </View>

        <View style={styles.footerBox3}>
          <View style={styles.footerBox1}>
            <View style={styles.textBox}>
              <Text style={styles.textFont1}>Trending Storylines</Text>
              <TouchableOpacity onPress={() => navigation.navigate('TrendingStoryline')}>
                <Text style={styles.textFont2}>View All</Text>
              </TouchableOpacity>
            </View>
          </View>
          <View style={styles.footerBox4}>
            <ScrollView horizontal>
              <View style={styles.postWidthtop}>
                <Storyline />
              </View>
              <View style={styles.postWidth}>
                <Storyline />
              </View>
              <View style={styles.postWidth}>
                <Storyline />
              </View>
              <View style={styles.postWidth}>
                <Storyline />
              </View>
            </ScrollView>
          </View>
        </View>

        <View style={styles.footerBox5}>
          <View style={styles.footerBox1}>
            <View style={styles.textBox}>
              <Text style={styles.textFont1}>All Storylines</Text>
              <View>
                <TouchableOpacity style={styles.filter} onPress={openTimeFilterPopUp}>
                  <FilterSVG />
                  <Text style={styles.filterFont}>Time Filter</Text>
                </TouchableOpacity>
                <TimeFilter
                  headerTitle="Time Filter"
                  isModalVisible={TimeFilterPopUpVisible}
                  getSelectedYears={captureYears}
                  getSelectedMonths={captureMonths}
                  onClose={closeTimeFilterPopUp}
                  onBtnPress={closeTimeFilterPopUp}
                />
              </View>
            </View>
          </View>
          <View style={styles.multipleEvents}>
            <StoryLineLongFormMultipleEvents
              carouselItems={[
                {
                  storyImage: images.storyImage,
                  previewText: 'Corbyn wins Labor conference Brexit vote in the latest polls',
                  commentText: '34K comments',
                  read: false,
                  onPressCard: () => {
                    pageNavigation('FullPost')
                  },
                  entityItems: [
                    {
                      image: images.iconStoryLineRelatedEntity1,
                      onPressEntity: () => {
                        console.log('Tap')
                        Entities2Navigation()
                      }
                    },
                    {
                      image: images.iconStoryLineRelatedEntity2,
                      onPressEntity: () => {
                        console.log('Tap2')
                        Entities2Navigation()
                      }
                    },
                    {
                      image: images.iconStoryLineRelatedEntity3,
                      onPressEntity: () => {
                        console.log('Tap3')
                        Entities2Navigation()
                      }
                    },
                    {
                      image: images.iconStoryLineRelatedEntity4,
                      onPressEntity: () => {
                        console.log('Tap4')
                        Entities2Navigation()
                      }
                    }
                  ]
                }
              ]}
            />
            <View style={styles.verticalgap} />
            <StoryLineLongFormMultipleEvents />
            <View style={styles.verticalgap} />
            <StoryLineLongFormMultipleEvents />
            <View style={styles.verticalgap} />
            <StoryLineLongFormMultipleEvents />
          </View>
        </View>
      </ScrollView>
    </View>
  )
}

Entities2.defaultProps = {
  title: 'Donald Trump',
  about: 'President of the USA',
  count: 54,
  details:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam.',
  profilePic: images.EntitiesImage11
}

export default Entities2
