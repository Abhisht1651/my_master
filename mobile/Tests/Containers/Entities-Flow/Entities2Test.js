import React from 'react'
import { shallow } from 'enzyme'
import { Image } from 'react-native'
import { TouchableOpacity, Text } from 'react-native'
import Entities2 from '../../../App/Containers/EntitiesScreens/Entities2'
// Test for Entities2 (snapshot and props)
describe('Entities2', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Entities2 route={{ params: { selected: [] } }} />) // no compile-time error
  })
  it('Entities2 component renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 5 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(5)
  })
  it('should render 11 Text', () => {
    expect(wrapper.find(Text)).toHaveLength(11)
  })
})
