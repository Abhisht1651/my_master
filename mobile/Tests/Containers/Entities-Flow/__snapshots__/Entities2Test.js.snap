// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entities2 Entities2 component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "elevation": 2,
        "flexDirection": "row",
        "height": 100.21428571428572,
        "justifyContent": "space-between",
        "paddingHorizontal": 50,
        "shadowColor": "#E6ECED",
        "width": "100%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
      />
    </ForwardRef>
    <Bubbles2 />
  </View>
  <ScrollViewMock>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "elevation": 1,
          "height": 847.7142857142857,
          "justifyContent": "center",
          "paddingBottom": 95.28571428571428,
          "paddingLeft": 50,
          "paddingRight": 62,
          "paddingTop": 60.785714285714285,
          "shadowColor": "#E6ECED",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 123.21428571428571,
            }
          }
        >
          <Profile
            profileIcons={
              Array [
                Object {
                  "default": "",
                },
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 80.5,
              "justifyContent": "center",
              "marginTop": 64.07142857142857,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Bold",
                "fontSize": 40,
                "letterSpacing": -0.28,
              }
            }
          >
            Donald Trump
          </Text>
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Light",
                "fontSize": 32,
                "letterSpacing": -0.22,
                "textAlign": "left",
              }
            }
          >
            President of the USA
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 93.64285714285714,
              "justifyContent": "center",
              "marginTop": 49.28571428571429,
            }
          }
        >
          <ForwardRef
            onPress={[Function]}
          >
            <X2SButton
              disabled={false}
              float={false}
              onPress={[Function]}
              outline={true}
              style={Object {}}
              title="Follow"
            />
          </ForwardRef>
          <Text
            style={
              Object {
                "color": "#8F8FAF",
                "fontFamily": "BananaGrotesk-Light",
                "fontSize": 28.000000000000004,
                "marginTop": 8.214285714285715,
              }
            }
          >
            54K Following
          </Text>
        </View>
        <View
          style={
            Object {
              "marginBottom": 95.28571428571428,
              "marginTop": 47.64285714285714,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 36,
                "letterSpacing": -0.24,
              }
            }
          >
            Details
          </Text>
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Light",
                "fontSize": 32,
                "letterSpacing": -0.22,
                "lineHeight": 44.35714285714286,
                "marginTop": 24.642857142857146,
              }
            }
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam.
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 302.2857142857143,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 138,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 50,
              "marginTop": 69,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 36,
                "letterSpacing": -0.24,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Similar Entities
          </Text>
          <ForwardRef
            onPress={[Function]}
          >
            <Text
              style={
                Object {
                  "color": "#7B46E4",
                  "fontFamily": "BananaGrotesk-Medium",
                  "fontSize": 32,
                  "letterSpacing": -0.18,
                  "opacity": 1,
                  "textAlign": "left",
                }
              }
            >
              View All
            </Text>
          </ForwardRef>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 179.07142857142858,
          }
        }
      >
        <ScrollViewMock
          horizontal={true}
        >
          <View
            style={
              Object {
                "paddingLeft": 26,
                "paddingRight": 8,
              }
            }
          >
            <ThumbsLarge
              profileIcons={
                Array [
                  Object {
                    "default": "",
                  },
                ]
              }
              title="India"
            />
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 8,
              }
            }
          >
            <ThumbsLarge
              profileIcons={
                Array [
                  Object {
                    "default": "",
                  },
                ]
              }
              title="USA"
            />
          </View>
          <View
            profileIcons={
              Array [
                Object {
                  "default": "",
                },
              ]
            }
            style={
              Object {
                "paddingHorizontal": 8,
              }
            }
          >
            <ThumbsLarge
              profileIcons={
                Array [
                  Object {
                    "default": "",
                  },
                ]
              }
              title="UNICEF"
            />
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 8,
              }
            }
          >
            <ThumbsLarge
              profileIcons={
                Array [
                  Object {
                    "default": "",
                  },
                ]
              }
              title="European"
            />
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 8,
              }
            }
          >
            <ThumbsLarge
              profileIcons={
                Array [
                  Object {
                    "default": "",
                  },
                ]
              }
              title="Europe"
            />
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 8,
              }
            }
          >
            <ThumbsLarge
              profileIcons={
                Array [
                  Object {
                    "default": "",
                  },
                ]
              }
              title="Default"
            />
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 8,
              }
            }
          >
            <ThumbsLarge
              profileIcons={
                Array [
                  Object {
                    "default": "",
                  },
                ]
              }
              title="Default"
            />
          </View>
        </ScrollViewMock>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 673.5714285714286,
        }
      }
    >
      <View
        style={
          Object {
            "height": 138,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 50,
              "marginTop": 69,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 36,
                "letterSpacing": -0.24,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            Trending Storylines
          </Text>
          <ForwardRef
            onPress={[Function]}
          >
            <Text
              style={
                Object {
                  "color": "#7B46E4",
                  "fontFamily": "BananaGrotesk-Medium",
                  "fontSize": 32,
                  "letterSpacing": -0.18,
                  "opacity": 1,
                  "textAlign": "left",
                }
              }
            >
              View All
            </Text>
          </ForwardRef>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 529,
            "width": "100%",
          }
        }
      >
        <ScrollViewMock
          horizontal={true}
        >
          <View
            style={
              Object {
                "marginLeft": 50,
                "width": 600,
              }
            }
          >
            <storyLineShortCardDefault
              border={true}
              followDisplay={true}
              followerDisplay={true}
              following={true}
              friendsFollowing="45 friends Following"
              heading="World"
              onPressFollow={[Function]}
              reach={45}
              storyImage={
                Object {
                  "default": "",
                }
              }
              storyTitle="UK exit from the European Union"
              trending={true}
              updatedTime="2m"
            />
          </View>
          <View
            style={
              Object {
                "marginLeft": 30,
                "width": 600,
              }
            }
          >
            <storyLineShortCardDefault
              border={true}
              followDisplay={true}
              followerDisplay={true}
              following={true}
              friendsFollowing="45 friends Following"
              heading="World"
              onPressFollow={[Function]}
              reach={45}
              storyImage={
                Object {
                  "default": "",
                }
              }
              storyTitle="UK exit from the European Union"
              trending={true}
              updatedTime="2m"
            />
          </View>
          <View
            style={
              Object {
                "marginLeft": 30,
                "width": 600,
              }
            }
          >
            <storyLineShortCardDefault
              border={true}
              followDisplay={true}
              followerDisplay={true}
              following={true}
              friendsFollowing="45 friends Following"
              heading="World"
              onPressFollow={[Function]}
              reach={45}
              storyImage={
                Object {
                  "default": "",
                }
              }
              storyTitle="UK exit from the European Union"
              trending={true}
              updatedTime="2m"
            />
          </View>
          <View
            style={
              Object {
                "marginLeft": 30,
                "width": 600,
              }
            }
          >
            <storyLineShortCardDefault
              border={true}
              followDisplay={true}
              followerDisplay={true}
              following={true}
              friendsFollowing="45 friends Following"
              heading="World"
              onPressFollow={[Function]}
              reach={45}
              storyImage={
                Object {
                  "default": "",
                }
              }
              storyTitle="UK exit from the European Union"
              trending={true}
              updatedTime="2m"
            />
          </View>
        </ScrollViewMock>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 138,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 50,
              "marginTop": 69,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 36,
                "letterSpacing": -0.24,
                "opacity": 1,
                "textAlign": "left",
              }
            }
          >
            All Storylines
          </Text>
          <View>
            <ForwardRef
              onPress={[Function]}
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": 176,
                }
              }
            >
              <SvgMock />
              <Text
                style={
                  Object {
                    "color": "#8F8FAF",
                    "fontSize": 24,
                    "letterSpacing": -0.14,
                  }
                }
              >
                Time Filter
              </Text>
            </ForwardRef>
            <TimeFilter
              getSelectedMonths={[Function]}
              getSelectedYears={[Function]}
              headerTitle="Time Filter"
              isModalVisible={false}
              onBtnPress={[Function]}
              onClose={[Function]}
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <StoryLineLongFormMultipleEvents
          active={false}
          carouselItems={
            Array [
              Object {
                "commentText": "34K comments",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "read": false,
                "storyImage": Object {
                  "default": "",
                },
              },
            ]
          }
          category="World"
          date="July 25th,1991"
          endStatus="Ended"
          followDisplay={true}
          following={false}
          friendsFollowing="241K Followers"
          onPressFollow={[Function]}
          onPressRaven={[Function]}
          onPressReact={[Function]}
          onPressShare={[Function]}
          onPressVoterIcons={[Function]}
          reach="45K Reached"
          storyTitle="UK exit from the European Union"
          totalStoryItems={8}
          trending={true}
          updatedTime="2m"
          voterIcons={
            Array [
              Object {
                "default": "",
              },
              Object {
                "default": "",
              },
              Object {
                "default": "",
              },
            ]
          }
        />
        <View
          style={
            Object {
              "height": 32.85714285714286,
            }
          }
        />
        <StoryLineLongFormMultipleEvents
          active={false}
          carouselItems={
            Array [
              Object {
                "commentText": "34K comments",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "read": false,
                "storyImage": Object {
                  "default": "",
                },
              },
              Object {
                "commentText": "34K comments",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "read": false,
                "storyImage": Object {
                  "default": "",
                },
              },
              Object {
                "commentText": "34K comments",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "read": false,
                "storyImage": Object {
                  "default": "",
                },
              },
              Object {
                "commentText": "",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "",
                "read": true,
                "storyImage": null,
              },
            ]
          }
          category="World"
          date="July 25th,1991"
          endStatus="Ended"
          followDisplay={true}
          following={false}
          friendsFollowing="241K Followers"
          onPressFollow={[Function]}
          onPressRaven={[Function]}
          onPressReact={[Function]}
          onPressShare={[Function]}
          onPressVoterIcons={[Function]}
          reach="45K Reached"
          storyTitle="UK exit from the European Union"
          totalStoryItems={8}
          trending={true}
          updatedTime="2m"
          voterIcons={
            Array [
              Object {
                "default": "",
              },
              Object {
                "default": "",
              },
              Object {
                "default": "",
              },
            ]
          }
        />
        <View
          style={
            Object {
              "height": 32.85714285714286,
            }
          }
        />
        <StoryLineLongFormMultipleEvents
          active={false}
          carouselItems={
            Array [
              Object {
                "commentText": "34K comments",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "read": false,
                "storyImage": Object {
                  "default": "",
                },
              },
              Object {
                "commentText": "34K comments",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "read": false,
                "storyImage": Object {
                  "default": "",
                },
              },
              Object {
                "commentText": "34K comments",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "read": false,
                "storyImage": Object {
                  "default": "",
                },
              },
              Object {
                "commentText": "",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "",
                "read": true,
                "storyImage": null,
              },
            ]
          }
          category="World"
          date="July 25th,1991"
          endStatus="Ended"
          followDisplay={true}
          following={false}
          friendsFollowing="241K Followers"
          onPressFollow={[Function]}
          onPressRaven={[Function]}
          onPressReact={[Function]}
          onPressShare={[Function]}
          onPressVoterIcons={[Function]}
          reach="45K Reached"
          storyTitle="UK exit from the European Union"
          totalStoryItems={8}
          trending={true}
          updatedTime="2m"
          voterIcons={
            Array [
              Object {
                "default": "",
              },
              Object {
                "default": "",
              },
              Object {
                "default": "",
              },
            ]
          }
        />
        <View
          style={
            Object {
              "height": 32.85714285714286,
            }
          }
        />
        <StoryLineLongFormMultipleEvents
          active={false}
          carouselItems={
            Array [
              Object {
                "commentText": "34K comments",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "read": false,
                "storyImage": Object {
                  "default": "",
                },
              },
              Object {
                "commentText": "34K comments",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "read": false,
                "storyImage": Object {
                  "default": "",
                },
              },
              Object {
                "commentText": "34K comments",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "Corbyn wins Labor conference Brexit vote in the latest polls",
                "read": false,
                "storyImage": Object {
                  "default": "",
                },
              },
              Object {
                "commentText": "",
                "entityItems": Array [
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                  Object {
                    "image": Object {
                      "default": "",
                    },
                    "onPressEntity": [Function],
                  },
                ],
                "onPressCard": [Function],
                "previewText": "",
                "read": true,
                "storyImage": null,
              },
            ]
          }
          category="World"
          date="July 25th,1991"
          endStatus="Ended"
          followDisplay={true}
          following={false}
          friendsFollowing="241K Followers"
          onPressFollow={[Function]}
          onPressRaven={[Function]}
          onPressReact={[Function]}
          onPressShare={[Function]}
          onPressVoterIcons={[Function]}
          reach="45K Reached"
          storyTitle="UK exit from the European Union"
          totalStoryItems={8}
          trending={true}
          updatedTime="2m"
          voterIcons={
            Array [
              Object {
                "default": "",
              },
              Object {
                "default": "",
              },
              Object {
                "default": "",
              },
            ]
          }
        />
      </View>
    </View>
  </ScrollViewMock>
</View>
`;
